cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/UnwindTest.scala

```diff
@@ -28,26 +28,76 @@ class UnwindTest extends DocumentingTestBase {
-      queryText = """UNWIND [1, 2, 3] AS x RETURN x""",
-      optionalResultExplanation = "Each value of the original list is returned as an individual row.",
-      assertions = (p) => assertEquals(List(1,2,3), p.columnAs[Int]("x").toList)
+      queryText = """UNWIND [1, 2, 3, null] AS x RETURN x, 'val' AS y""",
+      optionalResultExplanation = "Each value of the original list -- including `null` -- is returned as an individual row.",
+      assertions = (p) => assertEquals(List(Map("x" -> 1, "y" -> "val"), Map("x" -> 2, "y" -> "val"), Map("x" -> 3, "y" -> "val"), Map("x" -> null, "y" -> "val")), p.toList)
     )
   }
   @Test def distinct_list() {
     testQuery(
-      title = "Create a distinct list",
+      title = "Creating a distinct list",
       text = "We want to transform a list of duplicates into a set using `DISTINCT`.",
       queryText = """WITH [1, 1, 2, 2] AS coll UNWIND coll AS x WITH DISTINCT x RETURN collect(x) AS set""",
       optionalResultExplanation = "Each value of the original list is unwound and passed through `DISTINCT` to create a unique set.",
       assertions = (p) => assertEquals(List(List(1,2)), p.columnAs[Int]("set").toList)
     )
   }
 
+  @Test def concatenated_list() {
+    testQuery(
+      title = "Using `UNWIND` with any expression returning a list",
+      text = "Any expression that returns a list may be used with `UNWIND`.",
+      queryText =
+        """WITH [1, 2] AS a, [3, 4] AS b
+           UNWIND (a + b) AS x
+           RETURN x""",
+      optionalResultExplanation = "The two lists -- _a_ and _b_ -- are concatenated to form a new list, which is then operated upon by `UNWIND`.",
+      assertions = (p) => assertEquals(List(1, 2, 3, 4), p.columnAs[Int]("x").toList)
+    )
+  }
+
+  @Test def nested_lists() {
+    testQuery(
+      title = "Using `UNWIND` with a list of lists",
+      text = "Multiple `UNWIND` clauses can be chained to unwind nested list elements.",
+      queryText =
+        """WITH [[1, 2], [3, 4], 5] AS nested
+          UNWIND nested AS x
+          UNWIND x AS y
+          RETURN y""",
+      optionalResultExplanation = "The first `UNWIND` results in three rows for `x`, each of which contains an element of the original list (two of which are also lists); namely, `[1, 2]`, `[3, 4]` and `5`. " +
+        "The second `UNWIND` then operates on each of these rows in turn, resulting in five rows for `y`.",
+      assertions = (p) => assertEquals(List(1, 2, 3, 4, 5), p.columnAs[Int]("y").toList)
+    )
+  }
+
+  @Test def empty_list() {
+    testQuery(
+      title = "Using `UNWIND` with an empty list",
+      text = "Using an empty list with `UNWIND` will produce no rows, irrespective of whether or not any rows existed beforehand, or whether or not other values are being projected. " +
+        "Essentially, `UNWIND []` reduces the number of rows to zero, and thus causes the query to cease its execution, returning no results.",
```

Comment:
It is super valuable to explain the behaviour of `UNWIND []`, but it leads also to the question of why someone would ever do this. or they might as well not write the query at all. Of course, they would not write `UNWIND []`, but rather `UNWIND x` where x is some previous variable, or parameter that may or may not by an empty list at runtime. So it might be worth adding a little more to the description to explain how this feature can provide value.

Perhaps just adding to the end of the description: This has value in cases like `UNWIND v` where `v` is a variable from an earlier clause that may or may not be an empty list, and when it is an empty list this will behave just as a `MATCH` that has no results.

Reply:
👍